ste/downloader.go

```diff
@@ -43,17 +40,26 @@ type downloader interface {
 	Epilogue()
 }
 
+type folderDownloader interface {
+	SetFolderProperties(jptm IJobPartTransferMgr) error
+}
+
+//It seems the most simple option simple to say that all downloaders must implement all methods
+//Those that don't support folders should panic in SetFolderProperties (since it will never be called on them)
+type downloaderBase interface {
```

Comment:
Suggestion: Rather than making all downloaders implement un-necessary folder methods that just panic and serve no purpose, I suggest we reverse this and make it so folderDownloader includes fileDownloader, making it a logical extension to fileDownloader.

Sure, this means in our xferAnyToRemote and remoteToLocal, we have to type wrangle, BUT we can now add this panic to the xfer functions instead.

Reply:
Thanks!  Will do.